packages/hardhat-toolbox/src/index.ts

```diff
@@ -0,0 +1,29 @@
+import "@nomicfoundation/hardhat-chai-matchers";
+import "@nomicfoundation/hardhat-network-helpers";
+import "@nomiclabs/hardhat-ethers";
+import "@nomiclabs/hardhat-etherscan";
+import "@typechain/hardhat";
+import "hardhat-gas-reporter";
+import "solidity-coverage";
+
+import { extendConfig } from "hardhat/config";
+
+extendConfig((config, userConfig) => {
+  const configAsAny = config as any;
+
+  // hardhat-gas-reporter doesn't use extendConfig, so
+  // config.gasReporter === userConfig.gasReporter
+  // we use userConfig.gasReporter here for the types
+  const gasReporterConfig = userConfig.gasReporter;
+
+  configAsAny.gasReporter = gasReporterConfig ?? {};
+
+  if (gasReporterConfig?.enabled === undefined) {
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    configAsAny.gasReporter.enabled = process.env.REPORT_GAS ? true : false;
```

Comment:
`gasReporter` can be `undefined` here.

I couldn't cast config to any.

Reply:
That is handled by the previous line:

```js
configAsAny.gasReporter = gasReporterConfig ?? {};
```

The reason I'm casting the config to any is that hardhat-gas-reporter [doesn't extend the `UserConfig`](https://github.com/cgewecke/hardhat-gas-reporter/blob/8b3ad6d5b8d823faf1c1beee0b900cdc3a6335f2/src/type-extensions.ts), so `gasReporter` is not there at all.